src/BNO055.cpp

```diff
@@ -13,80 +13,69 @@ IMU::BNO055::BNO055(IO::I2C& i2C, uint8_t i2cSlaveAddress) : i2c(i2C) {
  * @return a boolean indicating whether or not the device was successfully initialized
  */
 bool IMU::BNO055::setup() {
-    // TODO: Check at the function call to see if it actually succeeds at setup.
-    log::LOGGER.log(log::Logger::LogLevel::INFO, "Starting Initialization...\r\n");
-
-    // First write an empty i2c command to the device.
-    uint8_t id = 0;
-    i2c.write(i2cAddress, &id, 0);
-    // After writing the empty command, we write a 0 to get the ID of the device, we then read the ID and store it in the id variable.
-    i2c.write(i2cAddress, 0x00);
-    i2c.read(i2cAddress, &id);
 
-    // Next we are writing the configuration bytes for the device. This sets the device into config mode which makes most of the
-    // registers writeable allowing us to configure its settings.
-    uint8_t configBytes[2] = {BNO055_OPR_MODE_ADDR, OPERATION_MODE_CONFIG};
-    i2c.write(i2cAddress, configBytes, 2);
-    EVT::core::time::wait(30);
+    log::LOGGER.log(log::Logger::LogLevel::INFO, "Starting Initialization...\r\n");
 
-    // We next send a system reset signal. This resets the device and brings it off the i2c network for period of time.
-    uint8_t resetBytes[2] = {BNO055_SYS_TRIGGER_ADDR, 0x20};
-    i2c.write(i2cAddress, resetBytes, 2);
```

Comment:
This entire chunk of code is not necessary since the device starts at configuration mode by default. It also makes no sense to reset it after it just booted up. In the new code, it first checks if i2c is activated and if so, then it resets the chip, otherwise it waits for i2c to boot up before continuing.